lib/cli/entry.js

```diff
@@ -33,38 +36,51 @@ module.exports = async (process, validateEngines) => {
     log.warn('cli', validateEngines.unsupportedMessage)
   }
 
-  let cmd
   // Now actually fire up npm and run the command.
   // This is how to use npm programmatically:
   try {
-    await npm.load()
+    const { exec } = await npm.load()
 
-    // npm -v
-    if (npm.config.get('version', 'cli')) {
-      output.standard(npm.version)
+    if (!exec) {
       return exitHandler()
     }
 
-    // npm --versions
-    if (npm.config.get('versions', 'cli')) {
-      npm.argv = ['version']
-      npm.config.set('usage', false, 'cli')
-    }
+    const command = npm.argv.shift()
+    const args = npm.argv
 
-    cmd = npm.argv.shift()
-    if (!cmd) {
+    if (!command) {
       output.standard(npm.usage)
       process.exitCode = 1
       return exitHandler()
     }
 
-    await npm.exec(cmd)
+    // Options are prefixed by a hyphen-minus (-, \u2d).
+    // Other dash-type chars look similar but are invalid.
+    const nonDashArgs = npm.argv.filter(a => /^[\u2010-\u2015\u2212\uFE58\uFE63\uFF0D]/.test(a))
+    if (nonDashArgs.length) {
+      log.error(
```

Comment:
Do we need to clean this? `npm install ‐‐registry=https://gar:password@npmjs.org`

Reply:
Pushed a commit which will redact these.